packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.tsx

```diff
@@ -178,14 +183,19 @@ export class ApplicationExtension<
                 )
             }
 
-            const component = componentMap[componentName]
-
-            if (!component) {
+            const Component = componentMap[componentName]
+            if (!Component) {
                 throw new Error(
                     `"${componentName}" was not found in the component map. Ensure that getComponentMap() includes a mapping for it in the ${this.getName()} extension`
                 )
             }
 
+            let component: React.ComponentType<any> = (props: Record<string, any>) => (
+                <Component {...props} {...route.componentProps} />
+            )
+            component = hoistNonReactStatics(component, Component)
+            component.displayName = componentName
```

Comment:
1. render the component with it's props
2. include any non-react statics like `getComponent()`
3. set the display name to what it was before serialization